src/models/tree/build_tree.cu

```diff
@@ -621,17 +629,14 @@ struct TreeBuilder {
                         legate::AccessorRO<double, 3> h,
                         NodeBatch batch)
   {
-    // TODO adjust kernel parameters dynamically
-    constexpr size_t elements_per_thread = 8;
-    constexpr size_t features_per_block  = 16;
-
-    const size_t blocks_x =
-      (batch.InstancesInBatch() + THREADS_PER_BLOCK * elements_per_thread - 1) /
-      (THREADS_PER_BLOCK * elements_per_thread);
-    const size_t blocks_y = (num_features + features_per_block - 1) / features_per_block;
-    dim3 grid_shape       = dim3(blocks_x, blocks_y, 1);
-    fill_histogram<TYPE, elements_per_thread, features_per_block>
-      <<<grid_shape, THREADS_PER_BLOCK, 0, stream>>>(X,
+    // warp kernel without additional caching / prefetching
+    const int threads_per_block = 256;
+    const size_t blocks_x = (batch.InstancesInBatch() + threads_per_block - 1) / threads_per_block;
+    const int features_per_warp = 64;
```

Comment:
Comment here might be nice on how you arrived at features_per_warp and what happens if its larger or smaller.